tools/python/openlr/quality.py

```diff
@@ -143,33 +148,36 @@ def parse_segments(tree, limit):
         # within limit are considered accurate, so golden path should be equal
         # matched path.
         golden_route = parse_route(s.find('GoldenRoute')) or matched_route
-        if not matched_route and not golden_route:
-            raise ValueError('At least one of golden route or matched route should be present')
-        try:
-            yield Segment(segment_id, matched_route, golden_route)
-        except:
-            print('Broken segment is {}'.format(segment_id))
-            raise
+        yield Segment(segment_id, golden_route, matched_route)
 
 def calculate(tree):
-    return {
-        s.segment_id: common_part(s.golden_route, s.matched_route)
-        for s in parse_segments(tree, args.limit)
-    }
+    result = {}
+    for s in parse_segments(tree, args.limit):
+        try:
+            result[s.segment_id] = common_part(s.golden_route, s.matched_route)
+        except AssertionError:
+            print('Something is wrong with segment {}'.format(s))
+            raise
+        except ValueError:
```

Comment:
Please describe the problem and maybe wrap in a new Exception with a clear message?

Reply:
Done.